mods/bones/init.lua

```diff
@@ -211,81 +275,73 @@ minetest.register_on_dieplayer(function(player)
 		return
 	end
 
-	local player_inv = player:get_inventory()
-	if is_all_empty(player_inv) then
-		minetest.log("action", player_name .. " dies at " .. pos_string ..
-			". No bones placed")
-		if bones_position_message then
-			minetest.chat_send_player(player_name, S("@1 died at @2.", player_name, pos_string))
-		end
-		return
+	local callback=function(stk)
+		transfer_stack_to_bones(stk,current_dead_player)
 	end
 
-	-- check if it's possible to place bones, if not find space near player
-	if bones_mode == "bones" and not may_replace(pos, player) then
-		local air = minetest.find_node_near(pos, 1, {"air"})
-		if air and not minetest.is_protected(air, player_name) then
-			pos = air
-		else
-			bones_mode = "drop"
-		end
+	for i=1,#dead_player_callbacks do
+		local fun=dead_player_callbacks[i]
+		fun(player,callback)
 	end
 
-	if bones_mode == "drop" then
-		for _, list_name in ipairs(player_inventory_lists) do
-			for i = 1, player_inv:get_size(list_name) do
-				drop(pos, player_inv:get_stack(list_name, i))
+	local bones_conclusion
+
+	if not ( current_dead_player.bones_pos ) then
+		drop(current_dead_player.player_pos, ItemStack("bones:bones"))
+		if not ( current_dead_player.dropped ) then
+			bones_conclusion="No bones placed"
+			if bones_position_message then
+				minetest.chat_send_player(player_name, S("@1 died at @2.", player_name, pos_string))
+			end
+		else
+			bones_conclusion="Inventory dropped"
+			if bones_position_message then
+				minetest.chat_send_player(player_name, S("@1 died at @2, and dropped their inventory.", player_name, pos_string, public_conclusion))
 			end
-			player_inv:set_list(list_name, {})
 		end
-		drop(pos, ItemStack("bones:bones"))
-		minetest.log("action", player_name .. " dies at " .. pos_string ..
-			". Inventory dropped")
-		if bones_position_message then
-			minetest.chat_send_player(player_name, S("@1 died at @2, and dropped their inventory.", player_name, pos_string))
+	else
+		if not ( current_dead_player.dropped ) then
+			bones_conclusion="Bones placed"
+			if bones_position_message then
+				minetest.chat_send_player(player_name, S("@1 died at @2, and bones were placed.", player_name, pos_string, public_conclusion))
+			end
+		else
+			bones_conclusion="Inventory partially dropped"
+			if bones_position_message then
+				minetest.chat_send_player(player_name, S("@1 died at @2, and partially dropped their inventory.", player_name, pos_string, public_conclusion))
```

Comment:
And here.

Reply:
> * the size should be determined based on how many items were added; the minimum size should be 4*8 to support 3d_armor's hack.

so do you want the armor to be dropped if there are no free slots in the main inventory? you can set the bones max slots to 64 to do so. or maybe I don't understand what do you mean.